### ISSUE TITLE
Load module parameters from @Parameter annotation

### ISSUE THREAD

**Issue by blcham** [kbss-cvut/s-pipes#219]
Many modules use @Parameter annotations but extends `AbstractModule` instead of `AnnotatedAbstractModule`. Thus they process parameters manually using `loadConfiguration` method.

Goal of this issue is to remove manual loading of modules wherever possible.

**Comment by palagdan** [comment]
@blcham 

f some types aren't annotated with the @Parameter annotation because they need to be configured manually, should I just leave them as they are? For example:

```java
@Override
    public void loadConfiguration() {
        rdf4jServerURL = getEffectiveValue(P_RDF4J_SERVER_URL).asLiteral().getString();
        rdf4jRepositoryName = getEffectiveValue(P_RDF4J_REPOSITORY_NAME).asLiteral().getString();
        try {
            rdf4jIgnoreIfExists = (Objects.equals(getEffectiveValue(P_RDF4J_IGNORE_IF_EXISTS).asLiteral().getString(), "true"));
        }
        catch (NullPointerException e){
            rdf4jIgnoreIfExists = false;
        }
        repositoryManager = new RemoteRepositoryManager(rdf4jServerURL);
    }
```

**Comment by blcham** [comment]
Yes, leave it as they are,  but if there is some parameter occurring multiple times, please mention it here so we can discuss how to extend @Parameter annotation to support it.

**Comment by palagdan** [comment]
@blcham What do you mean by a parameter occurring multiple times? Are you referring to a situation where the same annotation @parameter is used multiple times on one field?

**Comment by palagdan** [comment]
@blcham I have a question, if I remove some configuration and add parameter annotation instead, how can I test these changes?

**Comment by blcham** [comment]
I think it is done in [ApplyConstructModuleTest.executeSimple](https://github.com/kbss-cvut/s-pipes/blob/038b35026980cd230dae1fe8e3cab9b04382a440/s-pipes-core/src/test/java/cz/cvut/spipes/modules/ApplyConstructModuleTest.java#L31)

**Comment by blcham** [comment]
@palagdan

**Comment by palagdan** [comment]
@blcham 

I noticed that some variables are configured by overloading the getPropertyValue method, which accepts a Property and a literal default value to use if there is no object value for the current property. For example, the isReplace variable is configured as follows:
```java
 @Override
    public void loadConfiguration() {
        value = getEffectiveValue(SML.value);
        outputVariable = getStringPropertyValue(SM.outputVariable);
        isReplace = this.getPropertyValue(SML.replace, false);
    }
```
but loadConfiguration in [AnnotatedAbstractModule.java](https://github.com/kbss-cvut/s-pipes/blob/038b35026980cd230dae1fe8e3cab9b04382a440/s-pipes-core/src/main/java/cz/cvut/spipes/modules/AnnotatedAbstractModule.java) calls getEffectiveValue which does not accept a default value. This is why I cannot remove the manual configuration:
```java
RDFNode node = this.getEffectiveValue(ResourceFactory.createProperty(p.urlPrefix()+p.name()));
```

That is why I suggest to extend @Parameter annotation by one more attribute 'defaultValue'. Something like this:
```java

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface Parameter {
    String urlPrefix() default KBSS_MODULE.uri;
    String name();
    String comment();
    Object defaultValue();
}
```

With this change, we could assign a default value to an attribute if this.getEffectiveValue returns null and node is empty.

**Comment by blcham** [comment]
@palagdan I agree to add defaultValue() parameter there.

**Comment by palagdan** [comment]
@blcham 

I attempted to add handlers to the @Parameter annotation and use them for configuring attributes. However, I discovered that it’s more complex than simply adding some if-else statements to the loadConfiguration method in the AnnotatedAbstractModule class. Now, this method can call the appropriate getPropertyMethod in the AbstractModule class, whereas it previously handled only basic types (like int, boolean, etc.) and couldn't set RDFNode attributes.

The main issue is that some attributes, such as path variables, cannot be configured using the simple getPropertyValue() method. For example:
```java
 sourceFilePath = Optional.ofNullable(getEffectiveValue(SML.sourceFilePath))
                        .filter(RDFNode::isLiteral)
                        .map(RDFNode::asLiteral)
                        .map(Object::toString)
                        .map(s -> Paths.get(s))
                        .orElse(null);
```
I have a solution, but I’m not sure if it will meet your expectations. The idea is to create a separate method loadAnnotationConfiguration() for configuring attributes with annotations, and keep loadConfiguration() for classes that need to implement a method for attributes that cannot be configured via annotations. Because it much more complex to add another handler for attributes like this `sourceFilePath`.

Additionally, it is not possible to set a default value in the annotation with:
```java
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface Parameter {
    String urlPrefix() default KBSS_MODULE.uri;
    String name();
    String comment();
    Object defaultValue();
}
```
This is because Object defaultValue(); requires a specific type at compile time.
